src/test/java/net/sf/jabref/model/database/BibDatabaseTest.java

```diff
@@ -187,4 +188,29 @@ public void addSameStringIdTwiceThrowsKeyCollisionException() {
         database.addString(string);
         fail();
     }
+
+    @Test
+    public void testAddEntryEventReceivement() {
+        BibDatabase database = new BibDatabase();
+        BibEntry shouldBeEntry = new BibEntry();
+        TestEventListener tel = new TestEventListener();
+        database.registerListener(tel);
+        database.insertEntry(shouldBeEntry);
+        BibEntry isEntry = tel.getBibEntry();
+        assertEquals(shouldBeEntry, isEntry);
+    }
+
+    @Test
+    public void testRemoveEntryEventReceivement() {
+        BibDatabase database = new BibDatabase();
+        BibEntry shouldBeEntry = new BibEntry();
+        TestEventListener tel = new TestEventListener();
+        database.registerListener(tel);
+        database.insertEntry(shouldBeEntry);
+        tel = new TestEventListener();
```

Comment:
The problem here was to avoid already set `bibEntry` in `TestEventListener` while calling `removeEntry()`. Is we register it again, the same listener would be used again.